doc/whats_new/v1.1.rst

```diff
@@ -45,6 +45,10 @@ Changelog
   message suggests potential solutions.
   :pr:`21219` by :user:`Olivier Grisel <ogrisel>`.
 
+- |Enhancement| All scikit-learn models now generate a more informative
+  error message when setting invalid hyper-parameters with `set_params`.
+  :pr:`21542` by :user:`Olivier Grisel <ogrisel>`.
+
```

Comment:
That could go in 1.0.2, no ?

Reply:
I would rather only put bugfixes in bugfix releases.

Reply:
No problem :D 
```suggestion
- |Fix| All scikit-learn models now generate a more informative
  error message when setting invalid hyper-parameters with `set_params`.
  :pr:`21542` by :user:`Olivier Grisel <ogrisel>`.

```

Reply:
Jst kidding. I'm ok with that :)